bin/embark

```diff
@@ -4,11 +4,32 @@ try {
   eval('let __nodeTest = 123;');
 } catch(e) {
   if (e.name === 'SyntaxError') {
-    console.error("unsupported version of NodeJS. Make sure you are running nodejs 6.9.5 or above");
+    console.error("unsupported version of NodeJS. Make sure you are running nodejs 8.11.3 or above");
+
     process.exit();
   }
 }
 
-var Cmd = require('../cmd/cmd');
-var cli = new Cmd();
-cli.process(process.argv);
+function launchEmbark() {
+  var Cmd = require('../cmd/cmd');
+  var cli = new Cmd();
+  cli.process(process.argv);
+}
+
+const path = require('path');
+try {
+  const dappPackage = require(path.join(process.cwd(), 'package.json'));
+  require(path.join(process.cwd(), 'embark.json')); // Make sure we are in a Dapp
+  require('check-dependencies')(dappPackage, (state) => {
```

Comment:
Love it! Will adapt for [features/cli-shim](https://github.com/embark-framework/embark/tree/features/cli-shim).

It doesn't seem to trip on broken transitive dependencies, but I still think it's worthwhile.